tower-test/src/mock/mod.rs

```diff
@@ -2,35 +2,39 @@
 
 pub mod error;
 pub mod future;
+pub mod spawn;
 
-use crate::mock::{error::Error, future::ResponseFuture};
+use crate::mock::{error::Error, future::ResponseFuture, spawn::Spawn};
 use core::task::Waker;
 
 use tokio::sync::{mpsc, oneshot};
+use tower_layer::Layer;
 use tower_service::Service;
 
 use std::{
     collections::HashMap,
     future::Future,
-    pin::Pin,
     sync::{Arc, Mutex},
     task::{Context, Poll},
     u64,
 };
 
-/// Mock a task and Service.
-pub fn task_fn<T, U, F>(mut f: F)
+/// Spawn a layer onto a mock service.
+pub fn spawn_layer<T, U, L>(layer: L) -> (Spawn<L::Service>, Handle<T, U>)
 where
-    F: FnMut(&mut Context, &mut Pin<&mut Mock<T, U>>, &mut Pin<&mut Handle<T, U>>),
+    L: Layer<Mock<T, U>>,
 {
-    tokio_test::task::spawn(()).enter(|cx, _| {
-        let (mock, handle) = pair();
+    let (inner, handle) = pair();
+    let svc = layer.layer(inner);
```

Comment:
Reading this just made me realize that the `layer` method really should be called `before`. Would have read so much nicer. Oh well :man_shrugging:

Reply:
We can still change :P